tests/conftest.py

```diff
@@ -73,7 +84,7 @@ def datasets(tmpdir_factory):
     # Add two random null values to each column
     imax = len(df) - 1
     for col in df.columns:
-        if col in ["name-cat", "label"]:
+        if col in ["name-cat", "label", "id"]:
```

Comment:
The problem here is with adding nulls to CSV files . When reading in integer columns with nulls from CSV the column was getting interpreted as a float64 - and when the file didn't have nulls it got loaded as int64.  This caused a problem when both null entries ended up in one csv file: we read that one file in as float, and the next file as an int dtype - and this caused unittest errors when writing out later with pyarrow because the schema didn't match. I'm not sure why this used to work to be honest.